common/src/main/java/org/apache/falcon/hadoop/HadoopClientFactory.java

```diff
@@ -214,11 +215,19 @@ public FileSystem run() throws Exception {
      * @param executeUrl jt url or RM url
      * @throws IOException
      */
-    public void validateJobClient(String executeUrl) throws IOException {
+    public void validateJobClient(String executeUrl, String rmPrincipal) throws IOException {
         final JobConf jobConf = new JobConf();
         jobConf.set(MR_JT_ADDRESS_KEY, executeUrl);
         jobConf.set(YARN_RM_ADDRESS_KEY, executeUrl);
-
+        /**
+         * It is possible that the RM/JT principal can be different between clusters,
+         * for example, the cluster is using a different KDC with cross-domain trust
+         * with the Falcon KDC.   in that case, we want to allow the user to provide
+         * the RM principal similar to NN principal.
+         */
+        if (UserGroupInformation.isSecurityEnabled() && StringUtils.isNotEmpty(rmPrincipal)) {
```

Comment:
Usually StringUtils.isNotBlank() is used to make sure no empty string passed in for the property.